metrics-proxy/src/main/java/ai/vespa/metricsproxy/service/MetricsParser.java

```diff
@@ -28,6 +28,7 @@
  * @author Jo Kristian Bergum
  */
 public class MetricsParser {
+    private static final Double ZERO_DOUBLE = 0d;
```

Comment:
This shouldn't be necessary as long as `Double.valueOf()` is used

Reply:
Double.valueof does not have any of that as far as I can tell.
Long.valueof has for [-128, 127]. However when I use Long.valueof intellij says it can be dropped due to autoboxing.
Does that mean that auoboxing uses X.valueof, or is Intellij not correct. ?

Reply:
That is surprising. The Javadoc clearly states that `Double.valueOf()` should be used for caching/performance. I'm not sure what auto-boxing does. Th use of the `Double` constructor should be avoided though.